packages/orchestration/src/typeGuards.js

```diff
@@ -105,10 +108,27 @@ export const CosmosChainInfoShape = M.splitRecord(
   },
 );
 
+export const CosmosChainInfoOptionals = {
+  bech32Prefix: M.string(),
+  connections: M.record(),
+  stakingTokens: M.arrayOf({ denom: M.string() }),
+  // UNTIL https://github.com/Agoric/agoric-sdk/issues/9326
+  icqEnabled: M.boolean(),
+  pfmEnabled: M.boolean(),
+};
+
 /** @type {TypedPattern<ChainInfo>} */
-export const ChainInfoShape = M.splitRecord({
-  chainId: M.string(),
-});
+export const ChainInfoShape = M.splitRecord(
+  {
+    chainId: M.string(),
+    namespace: M.string(),
+    reference: M.string(),
+  },
+  {
+    cctpDestinationDomain: M.number(),
+    ...CosmosChainInfoOptionals,
```

Comment:
why make these all optional? Don't we want some guarantees that if a chain info is Cosmos that certain properties will be present?

Reply:
Endo Patterns do union with `M.or`.

A lot of `CosmosChainInfoOptionals` are optional but `bech32Prefix` isn't. Make it so the struct can be discriminated to know whether `bech32Prefix` must be present.